ert/data/record/_record.py

```diff
@@ -181,6 +183,99 @@ def __eq__(self, o: object) -> bool:
         return False
 
 
+RecordGen = TypeVar("RecordGen", BlobRecord, NumericalRecord)
+
+
+class RecordTree(Record, Generic[RecordGen]):
+    def __init__(self, record_dict: Dict[str, Any]) -> None:
+        self._record_type = RecordType.BYTES
+        self._flat_record_dict: Dict[str, RecordGen] = self._flatten_record_dict(
+            record_dict
+        )
+        try:
+            self._validate_data(self._flat_record_dict)
+        except BeartypeException as e:
+            raise RecordValidationError(str(e))
+        self._record_dict: Dict[str, Union[RecordGen, Mapping[str, Any]]] = record_dict
+
+    @abstractmethod
+    def _flatten_record_dict(
+        self, record_dict: Dict[str, Any], root: str = ""
+    ) -> Dict[str, RecordGen]:
+        pass
+
+    @beartype
+    def _validate_data(self, flat_record_dict: Dict[str, RecordGen]) -> None:
+        # beartype does not do deep validation on dicts, so we do non-nested validation
+        # TODO: remove once https://github.com/beartype/beartype/issues/53 is done
+        if flat_record_dict:
+            _record_type = list(flat_record_dict.values())[0].record_type
+            for val in flat_record_dict.values():
+                if val.record_type != _record_type:
+                    raise RecordValidationError(
+                        f"RecordTree needs same record types {_record_type}!={val.record_type}"
+                    )
+        else:
+            raise RecordValidationError("No records found in RecordTree")
+
+    @property
+    def data(self) -> record_data:
+        # internally, recordtree is represented as BlobRecord; we don't need data here
+        return json.dumps("RECORD_TREE").encode("utf-8")
+
+    @property
+    def record_type(self) -> RecordType:
+        return RecordType.BYTES
+
+    @property
+    def flat_record_dict(self) -> Dict[str, RecordGen]:
+        return self._flat_record_dict
+
+
+class BlobRecordTree(RecordTree[BlobRecord]):
+    def _flatten_record_dict(
+        self,
+        record_dict: Dict[str, Any],
+        root: str = "",
+    ) -> Dict[str, BlobRecord]:
+        _flat_record_dict: Dict[str, BlobRecord] = {}
```

Comment:
why the underscore at the beginning of this local variable name?

Reply:
will fix this too...